### PR TITLE
[SCB-1251] Reduce archaius callback

### PR DESCRIPTION
**Pull request by wujimin**
avoid register archaius callback for each priority property:
* callback is a CopyOnWriteArraySet, too many callback is not so good
* difficult to manage the unregister logic

add a manager to manage all PriorityProperty instances
add a inspectator module to view internal status of configuration mechanism, but not support sensitive words filter now.

screenshot

screenshot

### CODE REVIEW

**Comment by yhs0092** [comment]
For the memory leak problem caused by the callback of `DynamicProperty`, this PR solve it by abandoning the callback mechanism of `com.netflix.config.DynamicProperty` and using `org.apache.commons.configuration.event.ConfigurationListener` instead. Am I right?

**Comment by yhs0092** [comment]
Currently the register and unregister methods for `ConfigObject` and `PriorityProperty` are provided by `PriorityPropertyManager`, are they the recomended way to construct and destruct `ConfigObject` and `PriorityProperty`？
If so, is it better to change the constructor of `PriorityProperty` package private?

**Comment by wujimin** [comment]
> For the memory leak problem caused by the callback of `DynamicProperty`, this PR solve it by abandoning the callback mechanism of `com.netflix.config.DynamicProperty` and using `org.apache.commons.configuration.event.ConfigurationListener` instead. Am I right?

yes, you are right.
by this way, we only need to register one listener globally

**Comment by wujimin** [comment]
> `PriorityProperty`

this will cause too complex to manage ConfigObject, because if one field create failed, but previous PriorityProperty already register to manager, still need to unregister

**Comment by wujimin** [comment]
> > `PriorityProperty`
> 
> this will cause too complex to manage ConfigObject, because if one field create failed, but previous PriorityProperty already register to manager, still need to unregister

or i will provide another method to only create instance but not register it.

**Comment by yhs0092** [comment]
> 
> 
> > > `PriorityProperty`
> > 
> > 
> > this will cause too complex to manage ConfigObject, because if one field create failed, but previous PriorityProperty already register to manager, still need to unregister
> 
> or i will provide another method to only create instance but not register it.

The benifit of `priorityPropertyMap` and `priorityPropertyMap` is that by registering the config instances into these two map, we can notify them when the config values are changed, am I right? If so, the instances not registered cannot get updated dynamically, I think the attration of such instance is weak for our users. 
If the instances must be unregistered to avoid memory leak, I think it's better to add reminder in the comment of register methods.

By the way, I still don't get it why the ConfigObject instance need to be unregistered when part of it's fields PriorityProperty are failed to create. Currently it seems the register method of PriorityProperty is invoked nowhere out of PriorityPropertyManager, If the creation of a ConfigObject  instance failed, we can just abandon this instance and all of it's property can be GCed, as far as I see. Is there any details I'm not aware of?

**Comment by wujimin** [comment]
> > > > `PriorityProperty`
> > > 
> > > 
> > > this will cause too complex to manage ConfigObject, because if one field create failed, but previous PriorityProperty already register to manager, still need to unregister
> > 
> > 
> > or i will provide another method to only create instance but not register it.
> 
> The benifit of `priorityPropertyMap` and `priorityPropertyMap` is that by registering the config instances into these two map, we can notify them when the config values are changed, am I right? If so, the instances not registered cannot get updated dynamically, I think the attration of such instance is weak for our users.
> If the instances must be unregistered to avoid memory leak, I think it's better to add reminder in the comment of register methods.
> 
> By the way, I still don't get it why the ConfigObject instance need to be unregistered when part of it's fields PriorityProperty are failed to create. Currently it seems the register method of PriorityProperty is invoked nowhere out of PriorityPropertyManager, If the creation of a ConfigObject instance failed, we can just abandon this instance and all of it's property can be GCed, as far as I see. Is there any details I'm not aware of?

inside ConfigObjectFactory, if invoke org.apache.servicecomb.config.priority.PriorityPropertyManager#createPriorityProperty, then this PriorityProperty instance already registered to PriorityPropertyManager, so can not be GCed.

File: foundations/foundation-config/src/main/java/org/apache/servicecomb/config/priority/PriorityPropertyManager.java [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+      return;
+    }
+
+    if (keyCache == null) {
+      keyCache = new ConcurrentHashMapEx<>();
+      updateCache(priorityPropertyMap.values());
+      configObjectMap.values().stream().forEach(this::updateCache);
+    }
+
+    keyCache.getOrDefault(event.getPropertyName(), Collections.emptyList()).stream()
```
[the remark this answers is not in the document]
Comment (wujimin) [reply foundations/foundation-config/src/main/java/org/apache/servicecomb/config/priority/PriorityPropertyManager.java]: done

**Pull request merged**